core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -53,6 +55,7 @@ public Consumer(ConsumerAdapter rawAdapter, String topic, MsbConfigurations msbC
         this.clock = clock;
         this.channelMonitorAgent = channelMonitorAgent;
         this.validator = validator;
+        this.collectorManager = new CollectorManager(topic, channelManager);
```

Comment:
Now consumer has to know Collector and CollectorManager.
Also we have the very same problem we had before: for ResponderServer-aware consumers only messageHandler is used. Otherwise only collectorManager is used.